app/presenters/katello/candlepin_pools_presenter.rb

```diff
@@ -0,0 +1,41 @@
+module Katello
+  class CandlepinPoolsPresenter < SimpleDelegator
+    def initialize(pool)
+      super(OpenStruct.new(self.class.map_attributes(pool)))
+    end
+
+    def self.from_pools(pools)
+      pools.map do |pool|
+        self.new(pool)
+      end
+    end
+
+    def self.map_attributes(pool)
+      {
+        pool_id: pool['id'],
+        active: pool['activeSubscription'],
+        quantity: pool['quantity'],
+        start_date: pool['startDate'],
+        end_date: pool['endDate'],
+        contract_number: pool['contractNumber'],
+        consumed: pool['consumed'],
+        product_name: pool['productName'],
+        product_id: pool['productId'],
+        subscription_id: pool['subscriptionId']
+      }
+    end
+
+    def self.cp_request_params
+      { include: ['id',
+                  'activeSubscription',
+                  'quantity',
+                  'startDate',
+                  'endDate',
+                  'contractNumber',
+                  'consumed',
+                  'productName',
+                  'productId',
+                  'subscriptionId'] }
+    end
```

Comment:
Maybe i'm a stickler here, but a presenter is purely for display purposes and this method doesn't make much sense in one.

Reply:
I was looking at it from the perspective of: these are the columns this presenter expects from candlepin and knows how to display. Where else might you suggest putting it?

Reply:
I think moving it to a model layer that handles the backend call as well would be more ideal.  I'd prefer the controller 'call' into the model layer to fetch instances of the model rather than making those calls directly from the controller.  

For example, something like:
```
pools = Katello::UpstreamPool.fetch_pools(@organization, page_num, per_page)
respond(collection: {results: pools})
```